src/Monolog/Handler/SocketHandler.php

```diff
@@ -236,8 +236,31 @@ public function getChunkSize(): ?int
      */
     public function isConnected(): bool
     {
-        return is_resource($this->resource)
-            && !feof($this->resource);  // on TCP - other party can close connection.
+        /*
+         * We try to probe that we are not connected in order to return fast.
+         */
+        if (!is_resource($this->resource)) {
+            return false;
+        }
+
+        if (feof($this->resource)) {
+            return false;
+        }
+
+        $info = stream_get_meta_data($this->resource);
+
+        if (isset($info['timed_out']) && $info['timed_out']) {
+            return false;
+        }
+
+        $rStreams = [$this->resource];
+        $wStreams = null;
+
+        if (!stream_select($rStreams, $wStreams, $wStreams, (int)$this->getTimeout())) {
+            return false;
+        }
```

Comment:
I don't think this adds any value, it'll block for $timeout if there is no event to be read from the streams, but it doesn't necessarily mean it's timed out IMO if nothing happened on the stream.